server/db/mysql/adapter.go

```diff
@@ -42,11 +43,19 @@ const (
 	defaultMaxResults = 1024
 	// This is capped by the Session's send queue limit (128).
 	defaultMaxMessageResults = 100
+
+	mysqlConfigErrorText = "mysql config: dsn param is deprecated. Please, specify individual connection settings separately: https://pkg.go.dev/github.com/go-sql-driver/mysql#Config"
 )
 
 type configType struct {
-	DSN    string `json:"dsn,omitempty"`
-	DBName string `json:"database,omitempty"`
+	ms.Config
+	// Connection pool settings.
+	MaxOpenConns           int `json:"max_open_conns,omitempty"`
+	MaxIdleConns           int `json:"max_idle_conns,omitempty"`
+	ConnMaxLifetimeSeconds int `json:"conn_max_lifetime_seconds,omitempty"`
+	// Deprecated.
+	DSN      string `json:"dsn,omitempty"`
```

Comment:
You are hard-failing the initialization when the `DSN` is present (line 82 below). If so, why keep the DSN in the config at all? 

But I would give users a grace period: if `Config` is specified, ignore the `DSN`. Otherwise we are going to get a lot of complaints.

Reply:
Or fail only if both config and DSN are specified.

Reply:
Done.